R/RebuildPlot.R

```diff
@@ -1,635 +1,20 @@
-
-    Inc <- (Xvals[2] - Xvals[1]) / 2
-    for (II in 1:Npnt)
-    {
-      xx <- c(Xvals[II] - Inc, Xvals[II] - Inc, Xvals[II] + Inc, Xvals[II] + Inc)
-      yy <- c(0, Yvals[II], Yvals[II], 0)
-      polygon(xx, yy, col = "gray")
-    }
-    Yvals <- as.double(UUU[Ipnt:(Ipnt + Npnt - 1), 6]) * ymax * 1.2
-    lines(Xvals, Yvals, lty = 1, lwd = 5, col = "red")
-    title(Title)
-  }
-  #  ==================================================================================================
-  # make empty list
-  UUUs <- vector("list", 5)
-  # number of files to read
-  Nfiles <- length(fileN)
-  # if only 1 directory was input, repeat for each file
-  if (length(dirn) == 1) {
-    dirn <- rep(dirn, Nfiles)
-  }
-  # check to make sure number of directories matches number of files
-  if (length(dirn) != Nfiles) {
-    stop("length of 'dirn' should equal 1 or length of fileN:", Nfiles)
-  }
-  for (Ifile in 1:Nfiles)
-  {
-    FileName <- file.path(dirn[Ifile], fileN[Ifile])
-    message("FileName: ", FileName)
-    UUU <- read.table(
-      file = FileName, col.names = 1:ncols, fill = T,
-      colClasses = "character", comment.char = "$", sep = ","
-    )
-    UUUs[[Ifile]] <- UUU
-
-    # Extract key parameters
-    Nsim <- as.double(UUU[3, 1])
-    Amin <- as.double(UUU[4, 1])
-    Amax <- as.double(UUU[5, 1])
-    Nsex <- as.double(UUU[6, 1])
-    RecType <- as.double(UUU[7, 1])
-    Ncatch <- as.double(UUU[8, 1])
-    Ioutput <- as.double(UUU[9, 1])
-
-    # Plot Age versus fecudity
-    if (1 %in% Plots[[Ifile]]) Net_Spawn_Graph(UUU, Amin, Amax, Titles[Ifile])
-
-    # Recruitment and Recruits/Spawners
-    if (2 %in% Plots[[Ifile]]) RecruitmentPlots(UUU, Titles[Ifile])
-
-    # Histogram of B0
-    if (3 %in% Plots[[Ifile]]) B0Dist(UUU, Titles[Ifile])
-
-    # Histogram of recovery times
-    if (4 %in% Plots[[Ifile]]) RecHist(UUU, Titles[Ifile])
-
-    # Individual plots
-    if (6 %in% Plots[[Ifile]]) {
-      ind.list[[Ifile]] <- IndividualPlots(UUU, Titles[Ifile], yearmax)
-      message("Running IndividualPlots for file ", Ifile)
-    } else {
-      ind.list[[Ifile]] <- NULL
-    }
-
-    # First five trajectories of SSB/target
-    if (7 %in% Plots[[Ifile]]) FirstFive(UUU, Titles[Ifile], yearmax)
-
-    # Plot of when recovery occurs
-    if (8 %in% Plots[[Ifile]]) FinalRecovery(UUU, Titles[Ifile])
-  }
-
-  # Results across strategies
-  DoStrategies <- FALSE
-  for (Ifile in 1:Nfiles) {
-    if (5 %in% Plots[[Ifile]]) DoStrategies <- TRUE
-  }
-  if (DoStrategies == T) {
-    OutputList[["AltStrategies"]] <- AltStrategies(
-      fileN, UUUs, Options, "",
-      yearmax, Titles
-    )
-  }
-  if (pdf) dev.off()
-
-  OutputList[["ind.list"]] <- ind.list
-  return(invisible(OutputList))
+DoProjectPlots <- function(...) {
+  lifecycle::deprecate_stop(
+    when = "1.49.0",
+    what = "DoProjectPlots()",
+    details = "The function has been moved to https://github.com/pfmc-assessments/rebuilder."
```

Comment:
I suggest adding to the details that the function has also been renamed to `do_project_plots`.

Reply:
@chantelwetzel-noaa, good point.

Right now both `do_project_plots()` and `DoProjectPlots()` are present in the package and @brianlangseth-NOAA is planning additional cleanup. Therefore it seems safer for now to just point users to the rebuilder repo (and thus the README). Given that the "users" are just Brian at the moment, I'm not too worried about folks getting lost.

Reply:
I plan to remove do_project_plots() and keep DoProjectPlots() within the rebuilder repo. The latter was very slightly more up to date but otherwise they do the same thing. Doesn't look like that affects anything here